go-controller/pkg/kubevirt/pod.go

```diff
@@ -402,19 +376,56 @@ func filterNotComplete(vmPods []*corev1.Pod) []*corev1.Pod {
 	return notCompletePods
 }
 
-func IsMigrationReadyForTrafficHandoff(client *factory.WatchFactory, pod *corev1.Pod) (bool, error) {
-	vmPods, err := findVMRelatedPods(client, pod)
+type LiveMigrationState string
+
+const (
+	LiveMigrationInProgress        LiveMigrationState = "InProgress"
+	LiveMigrationTargetDomainReady LiveMigrationState = "TargetDomainReady"
+	LiveMigrationCompleted         LiveMigrationState = "Completed"
+)
+
+type LiveMigrationStatus struct {
+	SourcePod, TargetPod *corev1.Pod
+	State                LiveMigrationState
+}
+
+func DiscoverLiveMigrationStatus(client *factory.WatchFactory, pod *corev1.Pod) (*LiveMigrationStatus, error) {
+	vmKey := ExtractVMNameFromPod(pod)
+	if vmKey == nil {
+		return nil, nil
+	}
+
+	vmPods, err := client.GetPodsBySelector(pod.Namespace, metav1.LabelSelector{MatchLabels: map[string]string{kubevirtv1.VirtualMachineNameLabel: vmKey.Name}})
 	if err != nil {
-		return false, fmt.Errorf("failed finding related pods for pod %s/%s when checking live migration left overs: %v", pod.Namespace, pod.Name, err)
+		return nil, err
 	}
+
+	//TODO: cover failed migration
 	vmPods = filterNotComplete(vmPods)
 
-	if !podDuringMigration(vmPods) {
-		return false, nil
+	// no migration
+	if len(vmPods) < 2 {
+		return nil, nil
 	}
 
-	if isSourcePodAlreadyStale(pod, vmPods) && isTargetPodReady(getTargetPod(vmPods)) {
-		return true, nil
+	if len(vmPods) > 2 {
+		return nil, fmt.Errorf("unexpected live migration state at pods: %+v", vmPods)
 	}
-	return false, nil
+
+	// Sort vmPods by creation time
+	sort.Slice(vmPods, func(i, j int) bool {
+		// i less than j
+		return vmPods[j].CreationTimestamp.After(vmPods[i].CreationTimestamp.Time)
+	})
+
+	status := LiveMigrationStatus{
+		SourcePod: vmPods[0],
+		TargetPod: vmPods[1],
```

Comment:
could we trim down the objects ? IIUC we just require the labels on the pods. Right ?